lib/jekyll-github-metadata/site_github_munger.rb

```diff
@@ -35,14 +35,14 @@ def github_namespace
         when nil
           drop
         when Hash
-          Jekyll::Utils.deep_merge_hashes(drop, site.config["github"])
+          drop.merge(site.config["github"])
```

Comment:
The real fix seems to be this change, since [`#deep_merge_hashes` uses `merge!`](https://github.com/jekyll/jekyll/blob/ff60d51c9b59d3bb5cc89f6a4f493602b42c5cac/lib/jekyll/utils.rb#L321) which resolves all the keys in the `Drop`. Perhaps instead of this, we could modify the `Drop` to fall back to make a copy of the original `site.config["github"]` and use it instead of the computed Drop value. So when a user puts `site.github.repo` in their Liquid template, it queries the Drop, which first looks at `site.config["github"]["repo"]`, then at `Drop#repo` as a fallback. `Drop#repo` would then make the necessary API calls only if called.

Reply:
Maybe? But that's mostly what is happening with this approach, or so I am pretty sure.

`Drop.merge` uses [merge!](https://github.com/jekyll/jekyll/blob/ff60d51c9b59d3bb5cc89f6a4f493602b42c5cac/lib/jekyll/drops/drop.rb#L250) ultimately as well, so you are getting the same merge behavior, but instead of running it on every key from the drop, it gets run on every key from `site.config["github"]`.

And that merge puts everything in the [mutations](https://github.com/jekyll/jekyll/blob/ff60d51c9b59d3bb5cc89f6a4f493602b42c5cac/lib/jekyll/drops/drop.rb#L154) hash, and [checks that first](https://github.com/jekyll/jekyll/blob/ff60d51c9b59d3bb5cc89f6a4f493602b42c5cac/lib/jekyll/drops/drop.rb#L126) when the drop is queried.